module/documents/actor/actor.mjs

```diff
@@ -3061,7 +3064,7 @@ export default class Actor5e extends SystemDocumentMixin(Actor) {
         }));
         return;
       }
-      const prototypeTokenData = await baseActor.getTokenDocument();
+      const prototypeTokenData = (await baseActor.getTokenDocument()).toObject();
```

Comment:
Unrelated to the issue. Spotted it in passing. We should be using source data here, not prepared data.